### ISSUE TITLE
Stamp Microsoft.DotNet.MSBuildSdkResolver.dll with Git SHA

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/cli#10171]
I'm looking at Microsoft.DotNet.MSBuildSdkResolver.dll this version:

```
[assembly: AssemblyCompany ("Microsoft.DotNet.MSBuildSdkResolver")]
[assembly: AssemblyConfiguration ("Release")]
[assembly: AssemblyFileVersion ("2.2.100.0")]
[assembly: AssemblyInformationalVersion ("2.2.100-preview1-008633")]
[assembly: AssemblyProduct ("Sdk")]
[assembly: AssemblyTitle ("Microsoft.DotNet.MSBuildSdkResolver")]
[assembly: AssemblyVersion ("2.2.100.0")]
```

I'd like to find the source code to debug this .dll. I'm not sure if it's SourceLinked and/or if the .pdb is embedded.

It would be nice to stamp the Git SHA into the .dll as an assembly-level attribute so it's easy to pull the source code and check out the matching version.

Bonus: SourceLink the .pdb, make sure the .dll is deterministic and the .pdbs are portable and either published everywhere or embedded into the .dll.

**Comment by livarcocc** [comment]
@nguerrera to comment.

**Comment by nguerrera** [comment]
@livarcocc We should be doing  the "Bonus" part in the move of this repo to arcade, right? We have this already in dotnet/sdk last I checked.

@KirillOsenkov From the informational version, the sources are release/2.2.1xx branch at commit height 8633, which is rather out of date with 2.2.1xx. It will take me a bit of time to figure out how to reverse that into a SHA, but sharing that insight now in case you or someone else knows a fast way.

I'm just curious: Where did you end up with this build? We haven't inserted a 2.2.1xx resolver into Visual Studio.

**Comment by KirillOsenkov** [comment]
This is what VSMac's MSBuild use on a Mac. cc @akoeplinger @radical

I've found a branch that is close enough so that the debugger line numbers align reasonably well. So this is not blocking. Thanks!

**Comment by nguerrera** [comment]
By any chance are you investigating an issue that looks related to Microsoft/msbuild#3829 or https://github.com/OmniSharp/omnisharp-vscode/issues/2604#issuecomment-430330103?

It is looking like something broke in Microsoft.DotNet.MSBuildSdkResolver.dll x-plat.

**Comment by KirillOsenkov** [comment]
No, I was investigating https://github.com/mono/monodevelop/issues/6335, we've found the problem in VSMac (no support for SDK resolver manifest .xml files).

**Comment by KirillOsenkov** [comment]
Now I need to find the Git SHA for Microsoft.DotNet.MSBuildSdkResolver.dll version 2.2.202.0 :)

**Comment by nguerrera** [comment]
https://github.com/dotnet/cli/releases/tag/v2.2.202

https://github.com/dotnet/cli/commit/8a7ff6789d66a487bffc35aebb136621f5a1b3ee
